q2_amr/amrfinderplus/types/_format.py

```diff
@@ -0,0 +1,347 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2019-2023, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+from q2_types.feature_data import MixedCaseDNAFASTAFormat, ProteinFASTAFormat
+from qiime2.plugin import model
+
+
+class TextFormat(model.TextFileFormat):
+    def _validate_(self, level):
+        pass
+
+
+class BinaryFormat(model.BinaryFileFormat):
+    def _validate_(self, level):
+        pass
+
+
+class AMRFinderPlusDatabaseDirectoryFormat(model.DirectoryFormat):
```

Comment:
OK, so this is a little bit too wild 😅 What I propose here is:
1. rename all the variable names to by all lowercase - there is really no reason for them to reflect the file names exactly 1-to-1 and, by convention, we use all-uppercase to represent constants
2. you should group files with the same extension into `FileCollection`s - just adjust the regex for the file name so that it can recognize files for all the different organisms
3. I'm not sure whether it makes sense to have separate validation for all the tab-separated files? Do you find any of those particularly important? 😅

Reply:
First I also had the `FileCollection` but then there is no way to verify if all needed files are present.
I need 11 files with the extension nto for instance and with `FileCollection` it would be possible to import a database that has any number of files with the extension nto. Or is there a way to specify how many files there should be in one `FileCollection`?

And not really, they are all important.